bowling.rb

```diff
@@ -0,0 +1,35 @@
+score = ARGV[0] #スクリプト実行時に渡された引数を参照する
+scores = score.split(",") #点数を分割する
+shots = [] #メソッドの呼び出しを定義
+scores.each do |t|
+  if t == "X"
+    shots << 10
+    shots << 0
+  else
+    shots << t.to_i
+  end
+end
+
+frames = shots.each_slice(2).to_a do |t| # 配列内の要素を2つずつ渡す
+  frames << t
+end
+
+point = frames.each_with_index.sum do |frame, i| #フレーム数をカウントする
+  if i > 8 || frame.sum < 10
+    frame.sum
+  elsif frame[0] == 10
+    extra_point = frames[i.next]
+    frame_point = frame.sum + extra_point.sum
+    if extra_point[0] == 10
+      double_extra_point = frames[i.next.next] # ダブルの場合は20点と次の次のフレームの1投目を加算する
+      frame_point + double_extra_point[0]
+    else
+      frame_point
+    end
```

Comment:
ちなみに、この辺は後置ifにすれば
```suggestion
    frame_point += frames[i.next.next][0]  if extra_point[0] == 10
    frame_point
```

みたいに書けるので幾分すっきりするかもしれません。

Reply:
後置ifの発想は無かったです... 
こちらも訂正いたします。